### PR TITLE
Feature/update tslint

### PR DESCRIPTION
**Pull request by ofrogon**
Update TsLint and fix errors.

Update the TsLint rules base on existing ones and Angular-cli ones.

### CODE REVIEW

File: src/app/modal/modal-close/modal-close.directive.ts [hunk 1]
```diff
@@ -7,11 +7,11 @@ import { MzModalComponent } from '../modal.component';
...
-    this.modalComponent.close();
-  }
-
   constructor(
     private modalComponent: MzModalComponent,
   ) { }
+
+  @HostListener('click') onclick() {
+    this.modalComponent.close();
+  }
```
Comment (jfcere) [inline src/app/modal/modal-close/modal-close.directive.ts]: not sure i like the fact of putting @Hostlistener after the constructor ... I always liked to have those on top, getting a better overview with @input and @output of all bindings! Is it really a best practices to put them after the constructor?

File: src/app/validation/validation.component.ts [hunk 2]
```diff
@@ -82,6 +76,12 @@ export class MzValidationComponent implements AfterViewInit, OnDestroy {
     public renderer: Renderer,
   ) { }
 
+  @HostListener('focusout', ['$event.target'])
+  onFocusOut(target: Event) {
+    this.ngControl.control.markAsTouched();
+    this.setValidationState();
+  }
```
Comment (jfcere) [inline src/app/validation/validation.component.ts]: Same comment ...

File: src/app/modal/modal-close/modal-close.directive.ts (hunk 1 above)
Comment (ofrogon) [reply src/app/modal/modal-close/modal-close.directive.ts]: My bad...
In the latest Angular-cli (v1.7.2) the `member-ordering` order value is:

```json
"static-field",
"instance-field",
"static-method",
"instance-method"
```

That can translate to (as the schema here represent https://palantir.github.io/tslint/rules/member-ordering/ ; also after it's more a choice of what order you prefer `public`, `protected` and `private`):

```json
"public-static-field",
"protected-static-field",
"private-static-field",
"public-instance-field",
"protected-instance-field",
"private-instance-field",
"public-static-method",
"protected-static-method",
"private-static-method",
"public-instance-method",
"protected-instance-method",
"private-instance-method"
```

Not what I used yesterday (that consider constructor position and reverse some sections):

```json
"public-static-field",
"protected-static-field",
"private-static-field",
"public-static-method",
"protected-static-method",
"private-static-method",
"public-instance-field",
"protected-instance-field",
"private-instance-field",
"public-constructor",
"protected-constructor",
"private-constructor",
"public-instance-method",
"protected-instance-method",
"private-instance-method"
```

Overall, fixing the rule above allow to place `@HostListener` before `constructor`.

**Comment by jfcere** [comment]
I pretty much think the fact that the CI is failling is because you did not update yarn.lock file (run `yarn` and commit the lock file)

**Pull request merged**